src/app/components/app-status/app-status.component.ts

```diff
@@ -0,0 +1,17 @@
+import { Component } from '@angular/core';
+import { OpenAiService } from 'src/app/states/open-ai.service';
+
+@Component({
+  selector: 'app-status',
+  templateUrl: './app-status.component.html',
+  styleUrls: ['./app-status.component.scss'],
+})
+export class AppStatusComponent {
+  cost: string = '?';
+
+  constructor(private openAI: OpenAiService) {
+    this.openAI.totalCost.subscribe((cost) => {
```

Comment:
a unsubscribe would be missing and currently creates a "memory leak". either implement a unsubscribe in the ondestroy method or map the value in a pipe.

cost$ = this.openAI.totalCost.pipe(startWith(0), map(cost => cost.toFixed(2));
{{ cost$ | async }}